src/klein/app.py

```diff
@@ -329,12 +330,33 @@ def run(self, host, port, logFile=None):
 
         @param logFile: The file object to log to, by default C{sys.stdout}
         @type logFile: file object
+
+        @param proto: protocol string for endpoint, can be either "tcp" or "ssl"
+        @type proto: str
+
+        @param privateKey: filepath to private key for ssl
+        @type privateKey: str
+
+        @param certificate: filepath to certificate for ssl
+        @type certificate: str
+
+        @param dhParameters: filepath to dh parameters for ssl
+        @type dhParameters: str
         """
         if logFile is None:
             logFile = sys.stdout
 
         log.startLogging(logFile)
-        reactor.listenTCP(port, Site(self.resource()), interface=host)
+        server_string = "{}:{}:interface={}".format(proto, port, host)
+
+        if privateKey and certificate:
+            server_string += ":privateKey={}:certKey={}".format(privateKey,
+                                                                certificate)
+        if dhParameters:
+            server_string += ":dhParameters={}".format(dhParameters)
+
+        endpoint = endpoints.serverFromString(reactor, server_string)
+        endpoint.listen(Site(self.resource()))
```

Comment:
> On second thought I realized that one drawback of new method is that it will create two ways of doing same thing, so this may add some confusion.

I agree it would cause initial confusion, in the sense that ``TCP4ServerEndpoint`` or ``serverFromString`` causes general confusion in Twisted.  It can become a bit difficult to determine which function to use but at the end of the day you just pick one and move on to bigger problems.  With that being said, what would be the problem with having 2 methods?  One is a "dummy-proof" method where you supply the interface/port and a connection is started using ipv4.  The other is a method allows devs to utilize what ever protocol they want.  If that's not an acceptable solution, couldn't another method be added and then mark the ``Klein.run`` method for deprecation?